### PR TITLE
feature/mx-1702 improve errors and logs

### PR DESCRIPTION
**Pull request by cutoffthetop**
# PR Context
- some additional prep for mx-1702

# Changes
- mute warnings about labels used in queries but missing in graph
- split up search_merged_items_in_graph for better readability
- update cypher queries to use `CALL` clauses with correct variable scope
- BREAKING: drop support for neo4j server version 5.6 and lower

### CODE REVIEW

File: mex/backend/graph/cypher/merge_edges.cql [hunk 1]
```diff
@@ -35,7 +35,7 @@ CALL {
 <%- endif %>
 }
 WITH source, count(edge) as merged, collect(edge) as edges
-CALL {
+CALL (source, edges) {
     WITH source, edges
```
Comment (rababerladuseladim) [inline mex/backend/graph/cypher/merge_edges.cql:39]:
```suggestion
```

File: mex/backend/graph/cypher/merge_item.cql [hunk 2]
```diff
@@ -40,7 +40,7 @@ ON MATCH SET value_<<index>> += $nested_values[<<index>>]
 WITH current,
     [<<range(nested_edge_labels|count)|map("ensure_prefix", "edge_")|join(", ")>>] as edges,
     [<<range(nested_edge_labels|count)|map("ensure_prefix", "value_")|join(", ")>>] as values
-CALL {
+CALL (current, values) {
     WITH current, values
```
Comment (rababerladuseladim) [inline mex/backend/graph/cypher/merge_item.cql:44]:
```suggestion
```

File: mex/backend/graph/cypher/merge_edges.cql (hunk 1 above)
Comment (rababerladuseladim) [inline mex/backend/graph/cypher/merge_edges.cql:39]:
```suggestion
```

**Review by rababerladuseladim — CHANGES_REQUESTED** [review]
During local testing I get a bunch of errors like this: `FAILED tests/graph/test_connector.py::test_fetch_rule_items_empty - neo4j.exceptions.ConfigurationError: Notification filtering is not supported for the Bolt Protocol Version(5, 0). Server Agent 'None'`

Do I need to update my neo4j installation? And if yes: Could you communicate this in the dev channel and update our development setup guidelines?

**Comment by rababerladuseladim** [comment]
And while we're at it: What do you think about expanding the setup instructions in the README with the neo4j installation?

File: mex/backend/graph/cypher/merge_item.cql (hunk 2 above)
Comment (cutoffthetop) [reply mex/backend/graph/cypher/merge_item.cql]: fixed [a83d6c7](https://github.com/robert-koch-institut/mex-backend/pull/189/commits/a83d6c78b152bf862470cd1bf95084576c935abc)

File: mex/backend/graph/cypher/merge_edges.cql (hunk 1 above)
Comment (cutoffthetop) [reply mex/backend/graph/cypher/merge_edges.cql]: fixed [a83d6c7](https://github.com/robert-koch-institut/mex-backend/pull/189/commits/a83d6c78b152bf862470cd1bf95084576c935abc)

**Comment by cutoffthetop** [comment]
> Do I need to update my neo4j installation? And if yes: Could you communicate this in the dev channel and update our development setup guidelines?

the notification settings were introduced in neo4j 5.7, so if you were below that you'd had to upgrade. i added a BREAKING point to the changelog, and will post a reminder upon release.
i also added a bulletpoint on upgrading to our guidelines.

> And while we're at it: What do you think about expanding the setup instructions in the README with the neo4j installation?

our setup guide is quite specific to our needs, so i'd leave that in our internal docs. however, i added a short and more general version of it to the readme.

**Review by rababerladuseladim — DISMISSED** [review]
All good, merge away!

**Pull request merged**